Std/Tactic/TryThis.lean

```diff
@@ -72,69 +335,139 @@ partial def replaceMVarsByUnderscores [Monad m] [MonadQuotation m]
     let `(?$id:ident) := s | pure none
     if id.getId.hasNum || id.getId.isInternal then `(?_) else pure none
 
-/-- Delaborate `e` into an expression suitable for use in `refine`. -/
+/-- Delaborate `e` into syntax suitable for use by `refine`. -/
 def delabToRefinableSyntax (e : Expr) : TermElabM Term :=
   return ⟨← replaceMVarsByUnderscores (← delab e)⟩
 
+/-- Delaborate `e` into a suggestion suitable for use by `refine`. -/
+def delabToRefinableSuggestion (e : Expr) : TermElabM Suggestion :=
+  return { suggestion := ← delabToRefinableSyntax e, messageData? := e }
+
+/-! # Widget hooks -/
+
+/-- Core of `addSuggestion` and `addSuggestions`. Whether we use an inline display for a single
+element or a list display is controlled by `isInline`. -/
+private def addSuggestionCore (ref : Syntax) (suggestions : Array Suggestion)
```

Comment:
Should `addSuggestions` fail on an empty `suggestions` array, or should we leave that case to be handled by whoever's using it?

Reply:
I'm inclined to fail. Otherwise we'll be silently pretending to succeed in situations that should be failing.

Reply:
Ok, fixed in d9abcff. (I only touched `addSuggestions`, since we want to error before we log any info.)